sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,332 @@
+ */
+
+#ifndef _az_IOT_PNP_CLIENT_H
+#define _az_IOT_PNP_CLIENT_H
+
+#include <az_result.h>
+#include <az_span.h>
+
+// NOTE: az_iot_hub_client.h is included for internal use only.
+#include <az_iot_hub_client.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * @brief Azure IoT PnP Client options.
+ *
+ */
+typedef struct az_iot_pnp_client_options
+{
+  az_span user_agent; /**< The user-agent is a formatted string that will be used for Azure IoT
+                         usage statistics. */
+} az_iot_pnp_client_options;
+
+/**
+ * @brief Azure IoT PnP Client.
+ *
+ */
+typedef struct az_iot_pnp_client
+{
+  struct
+  {
+    az_iot_hub_client iot_hub_client;
+	az_span root_interface_name;
+  } _internal;
+} az_iot_pnp_client;
+
+/**
+ * @brief Gets the default Azure IoT PnP Client options.
+ * @details Call this to obtain an initialized #az_iot_pnp_client_options structure that can be
+ *          afterwards modified and passed to #az_iot_pnp_client_init.
+ *
+ * @return #az_iot_pnp_client_options.
+ */
+AZ_NODISCARD az_iot_pnp_client_options az_iot_pnp_client_options_default();
+
+/**
+ * @brief Initializes an Azure IoT PnP Client.
+ *
+ * @param[out] client The #az_iot_pnp_client to use for this call.
+ * @param[in] iot_hub_hostname The IoT Hub Hostname.
+ * @param[in] device_id The Device ID.
+ * @param[in] root_interface_name The root interface of the #az_iot_pnp_client.
+ * @param[in] options A reference to an #az_iot_pnp_client_options structure. Can be NULL.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_init(
+    az_iot_pnp_client* client,
+    az_span iot_hub_hostname,
+    az_span device_id,
+    az_span root_interface_name,
+    az_iot_pnp_client_options const* options);
+
+/**
+ * @brief Gets the MQTT user name.
+ *
+ * The user name will be of the following format:
+ * {iothubhostname}/{device_id}/?api-version=2018-06-30&{user_agent}?digital-twin-model-id={root_interface_name}
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
+ * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_user_name_get(
+    az_iot_pnp_client const* client,
+    az_span mqtt_user_name,
+    az_span* out_mqtt_user_name);
+
+/**
+ * @brief Gets the MQTT client id.
+ *
+ * The client id will be of the following format:
+ * {device_id}
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] mqtt_client_id An empty #az_span with sufficient capacity to hold the MQTT client id.
+ * @param[out] out_mqtt_client_id The output #az_span containing the MQTT client id.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_id_get(
+    az_iot_pnp_client const* client,
+    az_span mqtt_client_id,
+    az_span* out_mqtt_client_id);
+
+/**
```

Comment:
is this going to get pulled in as doxygen too? If you just want it to be a C comment remove the second asterisk

Reply:
fixing